src/traces/indicator/plot.js

```diff
@@ -0,0 +1,788 @@
+        var numbersbBox = m[1];
+        var translateY;
+        if(!trace._scaleNumbers) scaleRatio = 1;
+        if(trace._isAngular) {
+            // align vertically to bottom
+            translateY = numbersY - scaleRatio * numbersbBox.bottom;
+        } else {
+            // align vertically to center
+            translateY = numbersY - scaleRatio * (numbersbBox.top + numbersbBox.bottom) / 2;
+        }
+
+        // Stash the top position of numbersbBox for title positioning
+        trace._numbersTop = scaleRatio * (numbersbBox.top) + translateY;
+
+        return strTranslate(numbersX, translateY) + ' scale(' + scaleRatio + ')';
+    });
+}
+
+// Apply fill, stroke, stroke-width to SVG shape
+function styleShape(p) {
+    p
+        .style('fill', function(d) { return d.color;})
+        .style('stroke', function(d) { return d.line.color;})
+        .style('stroke-width', function(d) { return d.line.width;});
+}
+
+// Returns a tween for a transition’s "d" attribute, transitioning any selected
+// arcs from their current angle to the specified new angle.
+function arcTween(arc, endAngle, newAngle) {
+    return function() {
+        var interpolate = d3.interpolate(endAngle, newAngle);
+        return function(t) {
+            return arc.endAngle(interpolate(t))();
+        };
+    };
+}
+
+// mocks our axis
+function mockAxis(gd, opts, zrange) {
+    var fullLayout = gd._fullLayout;
+
+    var axisIn = {
+        type: 'linear',
+        ticks: 'outside',
+        range: zrange,
+        tickmode: opts.tickmode,
+        nticks: opts.nticks,
+        tick0: opts.tick0,
+        dtick: opts.dtick,
+        tickvals: opts.tickvals,
+        ticktext: opts.ticktext,
+        ticklen: opts.ticklen,
+        tickwidth: opts.tickwidth,
+        tickcolor: opts.tickcolor,
+        showticklabels: opts.showticklabels,
+        tickfont: opts.tickfont,
+        tickangle: opts.tickangle,
+        tickformat: opts.tickformat,
+        exponentformat: opts.exponentformat,
+        separatethousands: opts.separatethousands,
+        showexponent: opts.showexponent,
+        showtickprefix: opts.showtickprefix,
+        tickprefix: opts.tickprefix,
+        showticksuffix: opts.showticksuffix,
+        ticksuffix: opts.ticksuffix,
+        title: opts.title,
+        showline: true
+    };
+
+    var axisOut = {
+        type: 'linear',
+        _id: 'x' + opts._id
+    };
+
+    var axisOptions = {
+        letter: 'x',
+        font: fullLayout.font,
+        noHover: true,
+        noTickson: true
+    };
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(axisIn, axisOut, axisLayoutAttrs, attr, dflt);
+    }
+
+    handleAxisDefaults(axisIn, axisOut, coerce, axisOptions, fullLayout);
+    handleAxisPositionDefaults(axisIn, axisOut, coerce, axisOptions);
+
+    return axisOut;
+}
+
+function strTranslate(x, y) {
+    return 'translate(' + x + ',' + y + ')';
+}
+
+function strRotate(angle) {
```

Comment:
This thing is only used once. :hocho:

Reply:
Done in e6d8525ab07c56ca10bdbac6762b06d1d57bf830